File: A_Magnets.py
def main() -> None:
    number_of_magnets: int = int(input())
    number_of_groups: int = 0
    current_magnet: str = ""

    for _ in range(number_of_magnets):
        magnet = input()
        if magnet != current_magnet:
            current_magnet = magnet
            number_of_groups += 1

    print(number_of_groups)

if __name__ == "__main__":
    main()

File: A_Magnets.rs
use std::io::BufRead;

fn main() {
    let mut stdin = std::io::stdin().lock();
    let mut read_line = || {
        let mut line = String::new();
        stdin.read_line(&mut line).unwrap();
        line
    };

    let number_of_magnets: u32 = read_line().trim().parse().unwrap();
    let mut number_of_groups: u32 = 0;
    let mut current_magnet: String = "".to_string();

    for _ in 0..number_of_magnets {
        let magnet = read_line().trim().to_string();
        if magnet != current_magnet {
            current_magnet = magnet;
            number_of_groups += 1;
        }
    }

    println!("{}", number_of_groups);
}